### ISSUE TITLE
FPGA (Field-Programmable Gate Array) -> CIPP (Circuit Intégré Prédiffusé Programmable)

### ISSUE THREAD

**Issue by pacalet** [soulaklabs/bitoduc.fr#265]
Cette traduction est inexacte, je pense. Les FPGA n'ont jamais été des circuits prédiffusés. Ce sont des circuits intégrés complets auxquels il ne manque aucune couche et qu'on peut utiliser immédiatement sans en terminer la fabrication.

Je propose donc plutôt "_Circuit Intégré Reconfigurable (CIR)_".

Explication détaillée : il y a peut-être là une petite confusion avec la technologie, aujourd'hui presque abandonnée, des _Gate Arrays_ (mers de portes), sorte d'intermédiaire entre le circuit spécifique (_full custom_ ou _standard cell_) et le FPGA. Les mers de portes sont effectivement des circuits prédiffusés (ça s'appelle ainsi parce que seules les zones de diffusion des transistors et leurs grilles sont implantées dans le silicium) dont on termine la fabrication à la demande, ce qui permet de les personnaliser en interconnectant les transistors comme on veut grâce aux couches de polysilicium cristallin et de métal. En moyenne série c'est moins cher que les circuits spécifiques parce que la matrice prédiffusée est fabriquée en grande quantité et que des coûts d'ingénierie non-récurrents sont donc partagés entre de nombreux clients. Et c'est aussi moins cher que les FPGA car c'est plus petit. Mais en petit volume c'est plus cher que les FPGAs parce qu'il y a tout de même des étapes de fabrication spécifiques et, en gros volume, c'est plus cher que les circuits intégrés spécifiques parce que c'est plus gros.

Leur seule application actuelle, à ma connaissance, c'est la réduction du coût d'un produit stable à base de FPGA en remplaçant tout ou partie ces derniers par des mers de portes spécialisées pour la configuration précise du ou des FPGA qu'elle remplacent. On perd complétement la flexibilité mais on gagne un peu en coût et en performance. Les gros fabricants de FPGA (Xilinx, Intel/Altera...) proposent ce genre de service. Le cycle de vie d'un tel produit commence donc avec des FPGA jusqu'au moment ou le nombre de ventes annuelles rend la solution à base de mers de portes plus économique. Et si les volumes de vente continuent de croitre on finit par atteindre le point où un circuit spécifique devient plus intéressant que les mers de portes.

**Comment by quentin** [comment]
J’adhère à votre logique.

On perd la notion de _in situ_ (Field-Programmable) dans votre suggestion. De même que la notion de _mer de portes_.
Une amélioration est-elle possible?

**Comment by pacalet** [comment]
> J’adhère à votre logique.
> 
> On perd la notion de _in situ_ (Field-Programmable) dans votre suggestion. De même que la notion de _mer de portes_.
> Une amélioration est-elle possible?

Oui, on perd ces deux informations mais, là encore, je crois qu'elles sont un peu obsolètes. Les _mers de portes_ n'en sont plus vraiment car un FPGA contient aujourd'hui, en plus des cellules de logique, des blocs de mémoire, des accélérateur de traitement du signal numérique (_DSP blocks_), des synthétiseurs d'horloge, etc. Certains circuits embarquent même des micro-processeurs complets avec leurs périphériques principaux.

Donc si on souhaite une traduction qui reflète la situation actuelle "_Circuit Intégré Reconfigurable_ (CIR)" ou "_Circuit Intégré Flexible Reconfigurable_ (CIFR)" ne me semblent pas trop éloignés. Si on préfère une traduction fidèle au sens original - originel, on pourrait tenter _Matrice de Portes logique, Reconfigurable In-situ (MPRI)_ ou quelque chose de ce genre.

**Comment by quentin** [comment]
> _Matrice de Portes logique, Reconfigurable In-situ (MPRI)_

Bluffant! J'aime d'autant plus que l'on conserve 4 lettres à l'accronyme.

**Comment by p0nce** [comment]
Merci beaucoup @pacalet pour ces très bonnes contributions ! Gageons que d'autres sauront émuler votre réussite dans le jeu difficile de la contribution Bitoduc.

Il ne reste plus qu'à transformer l'essai.
